pkg/cloudprovider/controller/nodecontroller.go

```diff
@@ -437,6 +437,17 @@ func (s *NodeController) deletePods(nodeID string) error {
 		if err := s.kubeClient.Pods(pod.Namespace).Delete(pod.Name); err != nil {
 			glog.Errorf("Error deleting pod %v: %v", pod.Name, err)
 		}
+		if pod.Spec.RestartPolicy == api.RestartPolicyAlways {
```

Comment:
It's the job of replication controller to recreate pods. Node controller should not be doing that.